src/sentry/tasks/integrations/sync_status_outbound.py

```diff
@@ -31,8 +32,10 @@ def sync_status_outbound(group_id: int, external_issue_id: int) -> Optional[bool
         # Issue link could have been deleted while sync job was in the queue.
         return None
 
-    integration = Integration.objects.get(id=external_issue.integration_id)
-    installation = integration.get_installation(organization_id=external_issue.organization_id)
+    integration = integration_service.get_integration(integration_id=external_issue.integration_id)
+    installation = integration_service.get_installation(
+        integration_id=integration.id, organization_id=external_issue.organization_id
+    )
```

Comment:
Another place where fetching the integration, installation and orgintegration all at once could help.